### PR TITLE
[ `ForSequenceClassification`] Support `left` padding

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Update the computation of `sequence_lengths` that determines the index of the outpus to be pooled on models that use pooled inputs in such a way.
Adresses #24265

- [ ] add a common test to make sure whether you pad left or right, sequence outputs are the same

### CODE REVIEW

File: src/transformers/models/gpt_neox/modeling_gpt_neox.py [hunk 1]
```diff
@@ -926,7 +926,7 @@ def forward(
             sequence_lengths = -1
         else:
             if input_ids is not None:
-                sequence_lengths = (torch.ne(input_ids, self.config.pad_token_id).sum(-1) - 1).to(logits.device)
+                sequence_lengths = (torch.eq(input_ids, self.config.pad_token_id).argmax(-1)).to(logits.device)
```
Comment (ArthurZucker) [inline src/transformers/models/gpt_neox/modeling_gpt_neox.py:929]:
```suggestion
                sequence_lengths = (torch.eq(input_ids, self.config.pad_token_id).long().argmax(-1)).to(logits.device)
```

File: src/transformers/models/gpt_neox/modeling_gpt_neox.py [hunk 2]
```diff
@@ -926,7 +926,7 @@ def forward(
             sequence_lengths = -1
         else:
             if input_ids is not None:
-                sequence_lengths = (torch.ne(input_ids, self.config.pad_token_id).sum(-1) - 1).to(logits.device)
+                sequence_lengths = (torch.eq(input_ids, self.config.pad_token_id).long().argmax(-1)).to(logits.device)
```
Comment (ArthurZucker) [inline src/transformers/models/gpt_neox/modeling_gpt_neox.py:929]:
```suggestion
                sequence_lengths = (torch.eq(input_ids, self.config.pad_token_id).long().argmax(-1) - 1).to(
                    logits.device
                )
```

**Review by sgugger — APPROVED** [review]
Works for me, thanks!
